ui/packages/catalog-ui-search/src/main/webapp/component/metacard-interactions/metacard-interactions.tsx

```diff
@@ -0,0 +1,275 @@
+    leftIcon: 'fa fa-plus',
+    rightIcon: 'fa fa-chevron-down',
+    label: 'Add / Remove from List',
+  })
+
+const createResultActionsExportRegion = (model: Model) =>
+  PopoutView.createSimpleDropdown({
+    componentToShow: ExportActionsView,
+    dropdownCompanionBehaviors: {
+      navigation: {},
+    },
+    modelForComponent: model.first(),
+    leftIcon: 'fa fa-external-link',
+    rightIcon: 'fa fa-chevron-down',
+    label: 'Export as',
+  })
+
+const getRegionForSelector = (el: El, selector: any) =>
+  new Marionette.Region({ el: el.find(selector) })
+
+class InteractionsView extends React.Component<Props, State> {
+  constructor(props: Props) {
+    super(props)
+
+    this.state = {
+      model: props.model,
+    }
+  }
+
+  componentDidMount = () => {
+    this.appendCssIfNeeded(this.props.model, this.props.el)
+    getRegionForSelector(this.props.el, '.interaction-add').show(
+      createAddRemoveRegion(this.props.model)
+    )
+    getRegionForSelector(this.props.el, '.interaction-actions-export').show(
+      createResultActionsExportRegion(this.props.model)
+    )
+    this.props.extensions &&
+      getRegionForSelector(this.props.el, '.interaction-extensions').show(
+        this.props.extensions
+      )
+    const setState = (model: Model) => this.setState({ model: model })
+
+    const toggleIsBlacklisted = () =>
+      this.props.el.toggleClass(
+        'is-blacklisted',
+        isBlacklisted(this.props.model)
+      )
+
+    this.props.listenTo(
+      this.props.model,
+      'change:metacard>properties',
+      setState
+    )
+    this.props.listenTo(
+      user
+        .get('user')
+        .get('preferences')
+        .get('resultBlacklist'),
+      'add remove update reset',
+      toggleIsBlacklisted
+    )
+  }
+
+  appendCssIfNeeded = (model: Model, el: El) => {
+    const isMixed = model.reduce((accum: number, item: Result) => {
+      if (item.isRemote()) {
+        el.toggleClass('is-remote', true)
+        return ++accum
+      }
+      if (item.isWorkspace()) {
+        el.toggleClass('is-workspace', true)
+        return ++accum
+      }
+      if (item.isResource()) {
+        el.toggleClass('is-resource', true)
+        return ++accum
+      }
+      if (item.isRevision()) {
+        el.toggleClass('is-revision', true)
+        return ++accum
+      }
+      if (item.isDeleted()) {
+        el.toggleClass('is-deleted', true)
+        return ++accum
+      }
+    }, 0)
+
+    el.toggleClass('is-mixed', isMixed > 1)
+
+    const currentWorkspace = store.getCurrentWorkspace()
+    el.toggleClass('in-workspace', Boolean(currentWorkspace))
+    el.toggleClass('is-downloadable', isDownloadble(model))
+    el.toggleClass('is-multiple', model.length > 1)
+    el.toggleClass('is-routed', isRouted())
+    el.toggleClass('is-blacklisted', isBlacklisted(model))
+    el.toggleClass('has-location', hasLocation(model))
+  }
+
+  render = () =>
+    (this.state.model && (
+      <>
```

Comment:
Is this the new shorthand for `React.Fragment?`

Reply:
It's new-ish (has been around for a while now), but yes

Reply:
Yeah, in .tsx files (and maybe .jsx) you can use the shorthand.  It's only in js files where you'll need to use the long name for it.